### PR TITLE
Added updateJetCollection() function to jetTools.py and related changes to the PATJetUpdater and the pat::Jet class

### PR DESCRIPTION
**Pull request by ferencek**
This PR builds on top of #8288 and adds the updateJetCollection() function to jetTools.py. This function allows, in a user-friendly way, to update the jet energy corrections and re-run existing or add additional b-tag discriminators. It can also be used to add TagInfos to jets. It primarily targets updating pat::Jets stored in MiniAOD. In addition, users can manually add an arbitrary set of additional userFloats.

An example cfg file `PhysicsTools/PatAlgos/test/patTuple_updateJets_fromMiniAOD_cfg.py` was also added and included in PAT unit tests.

One caveat concerns added TagInfos. They cannot be successfully retrieved from pat::Jets stored in the existing MiniAOD datasets due to a bug in the `PATJetSlimmer` which was not clearing the `tagInfoLabels_` data member of the `pat::Jet` class when dropping the TagInfos from slimmed jets. This PR therefore also includes a required bugfix for the `PATJetSlimmer`.

@ahinzmann @imarches @gpetruc @arizzi

### CODE REVIEW

File: PhysicsTools/PatAlgos/python/producersLayer1/jetUpdater_cfi.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import FWCore.ParameterSet.Config as cms
 
-patJetsUpdated = cms.EDProducer("PATJetUpdater",
+updatedPatJets = cms.EDProducer("PATJetUpdater",
```
Comment (slava77) [inline PhysicsTools/PatAlgos/python/producersLayer1/jetUpdater_cfi.py]: Is this renaming invisible to users (affects only transient objects)?
If it is not, why change the name?

**Comment by slava77** [comment]
@cmsbuild please test

@ferencek please update the PR subject line to be more descriptive of changes in the PATJetUpdater and pat::Jet

File: PhysicsTools/PatAlgos/python/producersLayer1/jetUpdater_cfi.py (hunk 1 above)
Comment (ferencek) [inline PhysicsTools/PatAlgos/python/producersLayer1/jetUpdater_cfi.py]: Yes, with the updateJetCollection() function from jetTools.py, this becomes a transient collection. Also, the new name is more in line with the PAT jet naming convention where user-defined labels and postfixes come after "PatJets" while "updated" is more a prefix-type descriptor.

**Comment by ferencek** [comment]
@slava77, PR subject line updated

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/12863	b91cf04
- code changes look OK. They affect only miniAOD part.
- jenkins tests pass and comparisons with the baseline show no differences in monitored quantities
- comparisons of miniAOD products (by branch size) show small differences in slimmedJetsPuppi and slimmedJets (changes are expected; no further checks are made to see specific variable changes in my checks)
- local test of miniAOD from AOD using pu35@25ns ttbar events show no significant change in CPU

**Pull request merged**